File: Renaissance.AspNet/frontend/Renaissance.Web/Services/LanAccessStateService.cs
namespace Renaissance.Web.Services;

public class LanAccessStateService
{
    public string? Token { get; private set; }
    public DateTime? ExpiresAtUtc { get; private set; }

    public bool IsUnlocked =>
        !string.IsNullOrWhiteSpace(Token)
        && ExpiresAtUtc is not null
        && ExpiresAtUtc.Value > DateTime.UtcNow;

    public void ApplyUnlock(string token, DateTime expiresAtUtc)
    {
        Token = token;
        ExpiresAtUtc = expiresAtUtc;
    }

    public void Clear()
    {
        Token = null;
        ExpiresAtUtc = null;
    }
}

File: Renaissance.AspNet/frontend/Renaissance.Angular/src/app/core/services/lan-access-state.service.ts
import { Injectable } from '@angular/core';

@Injectable({ providedIn: 'root' })
export class LanAccessStateService {
    token: string | null = null;
    expiresAtUtc: string | null = null;

    get isUnlocked(): boolean {
        return !!this.token && !!this.expiresAtUtc && new Date(this.expiresAtUtc) > new Date();
    }

    applyUnlock(token: string, expiresAtUtc: string): void {
        this.token = token;
        this.expiresAtUtc = expiresAtUtc;
    }

    clear(): void {
        this.token = null;
        this.expiresAtUtc = null;
    }
}
